functests/1_performance/performance.go

```diff
@@ -340,8 +342,137 @@ var _ = Describe("[rfe_id:27368][performance]", func() {
+		return fmt.Errorf("spec HugePages field is different")
+	}
+
+	if specHugePages != nil {
+		if string(*specHugePages.DefaultHugePagesSize) != string(*v1Profile.Spec.HugePages.DefaultHugePagesSize) {
+			return fmt.Errorf("defaultHugePagesSize field is different [v1alpha1: %s, v1: %s]",
+				*specHugePages.DefaultHugePagesSize, *v1Profile.Spec.HugePages.DefaultHugePagesSize)
+		}
+
+		if (specHugePages.Pages == nil) != (v1Profile.Spec.HugePages.Pages == nil) ||
+			len(specHugePages.Pages) != len(v1Profile.Spec.HugePages.Pages) {
+			return fmt.Errorf("pages field is different [v1alpha1: %v, v1: %v]",
+				specHugePages.Pages, v1Profile.Spec.HugePages.Pages)
+		}
+
+		for i, v1alpha1Page := range specHugePages.Pages {
+			v1page := v1Profile.Spec.HugePages.Pages[i]
+			if string(v1alpha1Page.Size) != string(v1page.Size) ||
+				(v1alpha1Page.Node == nil) != (v1page.Node == nil) ||
+				(v1alpha1Page.Node != nil && *v1alpha1Page.Node != *v1page.Node) ||
+				v1alpha1Page.Count != v1page.Count {
+				return fmt.Errorf("pages field is different [v1alpha1: %v, v1: %v]",
+					specHugePages.Pages, v1Profile.Spec.HugePages.Pages)
+			}
+		}
+	}
+
+	if !reflect.DeepEqual(v1alpha1Profile.Spec.MachineConfigLabel, v1Profile.Spec.MachineConfigLabel) {
+		return fmt.Errorf("machineConfigLabel field is different [v1alpha1: %v, v1: %v]",
+			v1alpha1Profile.Spec.MachineConfigLabel, v1Profile.Spec.MachineConfigLabel)
+	}
+
+	if !reflect.DeepEqual(v1alpha1Profile.Spec.MachineConfigPoolSelector, v1Profile.Spec.MachineConfigPoolSelector) {
+		return fmt.Errorf("machineConfigPoolSelector field is different [v1alpha1: %v, v1: %v]",
+			v1alpha1Profile.Spec.MachineConfigPoolSelector, v1Profile.Spec.MachineConfigPoolSelector)
+	}
+
+	if !reflect.DeepEqual(v1alpha1Profile.Spec.NodeSelector, v1Profile.Spec.NodeSelector) {
+		return fmt.Errorf("nodeSelector field is different [v1alpha1: %v, v1: %v]",
+			v1alpha1Profile.Spec.NodeSelector, v1Profile.Spec.NodeSelector)
+	}
+
+	specRealTimeKernel := v1alpha1Profile.Spec.RealTimeKernel
+	if (specRealTimeKernel == nil) != (v1Profile.Spec.RealTimeKernel == nil) {
+		return fmt.Errorf("spec RealTimeKernel field is different")
+	}
+
+	if *specRealTimeKernel.Enabled != *v1Profile.Spec.RealTimeKernel.Enabled {
+		return fmt.Errorf("specRealTimeKernel field is different [v1alpha1: %t, v1: %t]",
+			*specRealTimeKernel.Enabled, *v1Profile.Spec.RealTimeKernel.Enabled)
+	}
+
+	if !reflect.DeepEqual(v1alpha1Profile.Spec.AdditionalKernelArgs, v1Profile.Spec.AdditionalKernelArgs) {
+		return fmt.Errorf("additionalKernelArgs field is different [v1alpha1: %v, v1: %v]",
+			v1alpha1Profile.Spec.AdditionalKernelArgs, v1Profile.Spec.AdditionalKernelArgs)
+	}
+
+	specNUMA := v1alpha1Profile.Spec.NUMA
+	if (specNUMA == nil) != (v1Profile.Spec.NUMA == nil) {
+		return fmt.Errorf("spec NUMA field is different")
+	}
+
+	if specNUMA != nil {
+		if *specNUMA.TopologyPolicy != *v1Profile.Spec.NUMA.TopologyPolicy {
```

Comment:
specNUMA.TopologyPolicy can be nil, the same true for v1Profile.Spec.NUMA.TopologyPolicy

Reply:
See above, at this point both values are null or not null in the same time
If null => equal, nothing to check
if not-null => safe to check them both

Reply:
Right, is a pointer by itself

Reply:
done